portfolio/src/main/java/com/google/sps/servlets/AddCommentsServlet.java

```diff
@@ -0,0 +1,90 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import java.io.IOException;
+import java.util.Date;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that adds a new comment to datastore */
+@WebServlet("/add-comments")
+public class AddCommentsServlet extends HttpServlet {
+
+  // POST parameters
+  private String username;
+  private String comment;
+  private String imageId;
```

Comment:
Can you explain why you decided to use member variables for these fields? This servlet doesn't need to maintain state between requests, so I think that these should be moved to the doPost() method and stored as local variables. It looks like you're updating state when doPost() is called, but I think it's better practice to not maintain any state at all.

Reply:
I used them like this to be able to use them in different functions without having to pass them as arguments, like global variables basically. But I see now that it would be better practice to pass them as arguments. I'll do the changes.